mempool/src/tests/test_framework.rs

```diff
@@ -0,0 +1,516 @@
+        let (mempool_client_sender, mempool_consensus_sender, mempool_notifications, mempool) =
+            setup_mempool(
+                config.clone(),
+                application_handles,
+                peer_metadata_storage.clone(),
+                &runtime,
+            );
+
+        MempoolNode {
+            node_id,
+            config,
+            network_ids,
+            peer_network_ids: network_id_mapping,
+            mempool,
+            runtime,
+            mempool_client_sender,
+            mempool_consensus_sender,
+            mempool_notifications,
+            inbound_handles,
+            outbound_handles,
+            other_inbound_handles: HashMap::new(),
+            peer_metadata_storage,
+        }
+    }
+}
+
+fn setup_mempool(
+    config: NodeConfig,
+    network_handles: Vec<ApplicationNetworkHandle<MempoolNetworkSender, MempoolNetworkEvents>>,
+    peer_metadata_storage: Arc<PeerMetadataStorage>,
+    runtime: &Runtime,
+) -> (
+    MempoolClientSender,
+    MempoolConsensusSender,
+    MempoolNotifier,
+    Arc<Mutex<CoreMempool>>,
+) {
+    let (sender, _subscriber) = futures::channel::mpsc::unbounded();
+    let (ac_endpoint_sender, ac_endpoint_receiver) = mpsc_channel();
+    let (consensus_sender, consensus_events) = mpsc_channel();
+    let (mempool_notifier, mempool_listener) =
+        mempool_notifications::new_mempool_notifier_listener_pair();
+
+    let mempool = Arc::new(Mutex::new(CoreMempool::new(&config)));
+    let vm_validator = Arc::new(RwLock::new(MockVMValidator));
+    let db_rw = Arc::new(RwLock::new(DbReaderWriter::new(MockDbReaderWriter)));
+    let db_ro = Arc::new(MockDbReaderWriter);
+
+    let mut event_subscriber = EventSubscriptionService::new(ON_CHAIN_CONFIG_REGISTRY, db_rw);
+    let reconfig_event_subscriber = event_subscriber.subscribe_to_reconfigurations().unwrap();
+
+    start_shared_mempool(
+        runtime.handle(),
+        &config,
+        mempool.clone(),
+        network_handles,
+        ac_endpoint_receiver,
+        consensus_events,
+        mempool_listener,
+        reconfig_event_subscriber,
+        db_ro,
+        vm_validator,
+        vec![sender],
+        peer_metadata_storage,
+    );
+
+    (
+        ac_endpoint_sender,
+        consensus_sender,
+        mempool_notifier,
+        mempool,
+    )
+}
+
+fn node_runtime(node_id: NodeId) -> Arc<Runtime> {
+    Arc::new(
+        tokio::runtime::Builder::new_multi_thread()
+            .thread_name(format!("node-{:?}", node_id))
+            .enable_all()
+            .build()
+            .expect("[test-framework] failed to create runtime"),
+    )
+}
+
+fn mpsc_channel<T>() -> (
+    futures::channel::mpsc::Sender<T>,
+    futures::channel::mpsc::Receiver<T>,
+) {
+    futures::channel::mpsc::channel(1_024)
+}
+
+pub fn test_transaction(seq_num: u64) -> TestTransaction {
+    TestTransaction::new(1, seq_num, 1)
+}
+
+pub fn test_transactions(start: u64, num: u64) -> Vec<TestTransaction> {
```

Comment:
nit: I think this would be more readable if it was start and end as opposed to start and num.

Reply:
Yeah, probably.  I'll change that since I had trouble with it earlier